snow/networking/benchlist/benchlist.go

```diff
@@ -117,38 +117,77 @@ func NewBenchlist(
 		duration:               duration,
 		maxPortion:             maxPortion,
 	}
-	benchlist.timer = timer.NewTimer(benchlist.update)
-	go benchlist.timer.Dispatch()
-	return benchlist, benchlist.metrics.Initialize(ctx.Registerer)
```

Comment:
Definitely odd that we initialized the metrics after kicking off the goroutine... Not a bug - but felt pretty close